### PR TITLE
Trim integration config values

### PR DESCRIPTION
**Pull request by mknoedel**
Trim integration all unmasked config values prior to executing an integration.

Masked values are skipped because they may contain spaces at the start or the end. If you think there is a good place to put that information in the code let me know and I will add it!

### CODE REVIEW

File: packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts [hunk 1]
```diff
@@ -50,30 +48,6 @@ export interface InstanceConfigurationData {
   logger: IntegrationLogger;
 }
 
-function createInstanceConfiguration(
```
Comment (mknoedel) [inline packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts:53]: Moved to a shared utils file

File: packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts [hunk 2]
```diff
@@ -32,13 +29,14 @@ import {
...
 import { LOCAL_INTEGRATION_INSTANCE } from '../instance';
+import { createInstanceConfiguration } from './utils/createIntegrationConfig';
 
 const brotliDecompress = promisify(zlib.brotliDecompress);
 
 jest.mock('fs');
 
 function sleep(ms: number) {
-  return new Promise((resolve) => {
+  return new Promise<void>((resolve) => {
```
Comment (mknoedel) [inline packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts:39]: TsLint error

**Review by austinkelleher — DISMISSED** [review]
Awesome change!

**Review by aiwilliams — APPROVED** [review]
Excellent, thanks @mknoedel! Also appreciate the comments about why some changes were made.

**Comment by aiwilliams** [comment]
Oh, almost forgot... the CHANGELOG.md update would be a great addition so that it travels with the commit(s)/PR.

**Comment by ndowmon** [comment]
Curious, I haven't been following the conversation on this one... why did we choose to trim string values instead of having customers fix their configurations?

**Comment by austinkelleher** [comment]
@ndowmon It's pretty common for customers to make mistakes like that. There isn't a great reason to _not_ trim values imo. We do this other places in the backend as well.

**Review by austinkelleher — APPROVED** [review]

**Pull request merged**